front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts

```diff
@@ -0,0 +1,137 @@
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  deleteDataSource,
+  MANAGED_DS_DELETABLE_AS_BUILDER,
+} from "@app/lib/api/data_sources";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import { apiError } from "@app/logger/withlogging";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<void>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const user = auth.user();
+
+  if (!owner || !plan || !user) {
```

Comment:
I think you had it before, but we actually need to check for `auth.isUser`.

Reply:
added back